### PR TITLE
Added password authentication in MQTT

### PR DESCRIPTION
**Pull request by lprashant-94**

### CODE REVIEW

File: Edge/edgegateway_mqtt.c [hunk 1]
```diff
@@ -50,16 +51,21 @@ int connect_datonis_instance(char *server) {
 
     sprintf(client_id, "%s%0.0f", CLIENT_ID_PREFIX, get_time_ms());
 
+    get_hmac(configuration.access_key, password);
+    printf("Connecting to Datonis with credentials \n Username %s and Password %s" , configuration.access_key, password);
```
Comment (rajesh-altizon) [inline Edge/edgegateway_mqtt.c]: Never log the password.

File: Edge/edgegateway_mqtt.c [hunk 2]
```diff
@@ -36,12 +36,13 @@ static struct thing *registered_things[10];
 static int registered_thing_counter = 0;
 static char http_ack_context[65];
 static char client_id[50];
+static char password[65];
```
Comment (rajesh-altizon) [inline Edge/edgegateway_mqtt.c:39]: Let's make sure this much space is ok

**Review by rajesh-altizon — CHANGES_REQUESTED** [review]

File: Edge/edgegateway_mqtt.c (hunk 2 above)
Comment (lprashant-94) [reply Edge/edgegateway_mqtt.c:39]: Yeah this much space is ok for hmac, we used same space for events as well. SHA256 need 64 chars and one extra char for \0

**Comment by lprashant-94** [comment]
Review changes are done.

**Review by rajesh-altizon — APPROVED** [review]

**Pull request merged**